lib/Sema/TypeCheckAttr.cpp

```diff
@@ -597,9 +597,12 @@ void AttributeEarlyChecker::visitLazyAttr(LazyAttr *attr) {
   if (VD->isLet())
     diagnoseAndRemoveAttr(attr, diag::lazy_not_on_let);
 
+  auto attrs = VD->getAttrs();
   // 'lazy' is not allowed to have reference attributes
-  if (VD->getAttrs().hasAttribute<ReferenceOwnershipAttr>())
-    diagnoseAndRemoveAttr(attr, diag::lazy_not_strong);
+  if (attrs.hasAttribute<ReferenceOwnershipAttr>()) {
+    auto *refAttr = attrs.getAttribute<ReferenceOwnershipAttr>();
```

Comment:
You can collapse this with `if (auto *refAttr = attrs.getAttribute<ReferenceOwnershipAttr>())`, which avoids searching twice.